distributed/worker.py

```diff
@@ -1265,32 +1265,33 @@ async def heartbeat(self) -> None:
                     if hasattr(extension, "heartbeat")
                 },
             )
-
             end = time()
             middle = (start + end) / 2
-
             self._update_latency(end - start)
 
             if response["status"] == "missing":
-                # Scheduler thought we left. Reconnection is not supported, so just shut down.
-                logger.error(
-                    f"Scheduler was unaware of this worker {self.address!r}. Shutting down."
-                )
-                # Something is out of sync; have the nanny restart us if possible.
-                await self.close(nanny=False)
+                # Scheduler thought we left.
+                # This happens after one calls Scheduler.remove_worker(close=False).
+                # ***DO NOT call close()!***
+                # Read: https://github.com/dask/distributed/pull/8522
+                logger.info("Stopping heartbeat to the scheduler")
+                self.periodic_callbacks["heartbeat"].stop()
                 return
 
-            self.scheduler_delay = response["time"] - middle
             self.periodic_callbacks["heartbeat"].callback_time = (
                 response["heartbeat-interval"] * 1000
             )
+            self.scheduler_delay = response["time"] - middle
             self.bandwidth_workers.clear()
             self.bandwidth_types.clear()
+
         except OSError:
-            logger.exception("Failed to communicate with scheduler during heartbeat.")
-        except Exception:
-            logger.exception("Unexpected exception during heartbeat. Closing worker.")
-            await self.close()
```

Comment:
To the best of my knowledge this should be impossible; if it happens an unclosed worker is the least of the problems.

Reply:
The least unlikely use case for this is a weird edge case in the network stack; in which case we probably don't want to kill off the worker.

Reply:
This is not a good argument to remove this

Reply:
Added it back.